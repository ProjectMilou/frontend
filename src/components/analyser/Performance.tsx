import React from 'react';
import NumberFormat from 'react-number-format';
import { makeStyles, Theme } from '@material-ui/core';

export type PerformanceProps = {
  value: number;
  size?: string;
};

const useStyles = makeStyles<Theme, PerformanceProps, string>({
  performance: {
    fontSize: (props) => props.size || '24px',
    color: (props) => {
      switch (Math.sign(props.value)) {
        case -1:
          return '#B80C09';
        case 1:
          return '#3da97e';
        default:
          return '#b2b2b2';
      }
    },
  },
});

const Performance: React.FC<PerformanceProps> = (props) => {
  const { value } = props;
  const classes = useStyles(props);

  return (
    <NumberFormat
      value={value}
      displayType="text"
      suffix="%"
      decimalScale={2}
      fixedDecimalScale
      decimalSeparator="."
      className={classes.performance}
    />
  );
};

export default Performance;
