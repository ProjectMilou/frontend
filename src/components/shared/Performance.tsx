import React from 'react';
import NumberFormat from 'react-number-format';
import { makeStyles, Theme } from '@material-ui/core';

export type PerformanceProps = {
  value: number;
  size?: string;
};
// TODO change hard coded colors to palette
const useStyles = makeStyles<Theme, PerformanceProps, string>({
  performance: {
    fontSize: (props) => props.size || '24px',
    color: (props) => {
      switch (Math.sign(props.value)) {
        case -1:
          return '#B80C09';
        case 1:
          return '#3da97e';
        default:
          return '#b2b2b2';
      }
    },
  },
});

const Performance: React.FC<PerformanceProps> = (props) => {
  const { value } = props;
  const classes = useStyles(props);

  return (
    <NumberFormat
      value={value}
      displayType="text"
      thousandSeparator="."
      suffix=" %"
      decimalScale={2}
      fixedDecimalScale
      decimalSeparator=","
      className={classes.performance}
    />
  );
};

export default Performance;
