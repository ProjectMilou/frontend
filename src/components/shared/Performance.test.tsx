import { render } from '@testing-library/react';
import * as React from 'react';
import Performance, { PerformanceProps } from './Performance';

describe('Performance', () => {
  const defaultProps: PerformanceProps = {
    value: 13.37,
  };

  const renderComponent = (newProps?: Partial<PerformanceProps>) => {
    const props = { ...defaultProps, ...newProps };
    return {
      ...render(<Performance {...props} />),
      props,
    };
  };

  test('format performance', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test('always print decimal places', () => {
    const { container } = renderComponent({ value: 1337 });
    expect(container).toMatchSnapshot();
  });

  test('only print 2 decimal places', () => {
    const { container } = renderComponent({ value: -13.371234 });
    expect(container).toMatchSnapshot();
  });
});
