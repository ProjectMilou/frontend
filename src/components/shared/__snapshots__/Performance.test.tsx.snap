// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Performance always print decimal places 1`] = `
<div>
  <span
    class="makeStyles-performance-3 makeStyles-performance-4"
  >
    1.337,00 %
  </span>
</div>
`;

exports[`Performance format performance 1`] = `
<div>
  <span
    class="makeStyles-performance-1 makeStyles-performance-2"
  >
    13,37 %
  </span>
</div>
`;

exports[`Performance only print 2 decimal places 1`] = `
<div>
  <span
    class="makeStyles-performance-5 makeStyles-performance-6"
  >
    -13,37 %
  </span>
</div>
`;
